aggregator/aggregator.go

```diff
@@ -428,14 +427,45 @@ func (agg *aggregator) checkMetricType(mu unaggregated.MetricUnion) error {
 	}
 }
 
-func (agg *aggregator) ownedShards() []*aggregatorShard {
-	agg.RLock()
-	ownedShards := make([]*aggregatorShard, len(agg.shardIDs))
-	for i, shardID := range agg.shardIDs {
-		ownedShards[i] = agg.shards[shardID]
+func (agg *aggregator) ownedShards() (owned, toClose []*aggregatorShard) {
+	agg.Lock()
+	defer agg.Unlock()
+
+	owned = make([]*aggregatorShard, 0, len(agg.shardIDs))
+	for i := 0; i < len(agg.shardIDs); i++ {
+		shardID := agg.shardIDs[i]
+		// NB(xichen): if traffic has been cut off for a shard, the aggregator actively
+		// closes it to avoid wasting CPU cycles flushing its metric lists.
+		if shard := agg.shards[shardID]; !shard.IsCutoff() {
+			owned = append(owned, shard)
```

Comment:
nit: perhaps add a `continue` at the end of the if clause so that we don't need the else clause?

Reply:
Actually I feel it's more clear this way, would prefer to keep it as is unless you feel strongly.

Reply:
Don't feel strongly at all, I think `continue` is arguably more idiomatic is the only reason I mentioned it.